pathspider/plugins/evilbit.py

```diff
@@ -0,0 +1,111 @@
+    #version = pathspider.__version__
+    chains = [BasicChain, DNSChain, TCPChain, EvilChain]
+    connect_supported = ["tcpsyn", "dnsudp"]
+    packets = 2
+
+    def forge(self, job, seq):
+        sport = 0
+        while sport < 1024:
+            sport = int(RandShort())
+        if self.args.connect == 'tcpsyn':
+            l4 = (TCP(sport=sport, dport=job['dp']))
+        if self.args.connect == 'dnsudp':
+            l4 = (UDP(sport=sport, dport=job['dp']) /
+                  DNS(qd=DNSQR(qname=job['domain'])))
+        if ':' in job['dip']:
+            ip = IPv6(src=self.source[1], dst=job['dip'])
+        else:
+            ip = IP(src=self.source[0], dst=job['dip'])
+        if seq == 1:
+            ip.flags = 'evil'
+        return ip/l4
+
+    def combine_flows(self, flows):
+        for flow in flows:
+            if not flow['observed']:
+                return ['pathspider.not_observed']
+
+        conditions = []
+
+        if self.args.connect == 'tcpsyn':
+            conn0 = False
+            conn1 = False
+
+            if flows[0]['tcp_synflags_rev'] is not None and flows[0][
+                    'tcp_synflags_rev'] & TCP_SA == TCP_SA:
+                conn0 = True
+            if flows[1]['tcp_synflags_rev'] is not None and flows[1][
+                    'tcp_synflags_rev'] & TCP_SA == TCP_SA:
+                conn1 = True
+            if conn0 is True and conn1 is True:
+                conditions.append('evilbit.connectivity.works')
+                if flows[1]['evil_bit_syn_rev']:
+                    conditions.append('evilbit.mark.seen')
+                else:
+                    conditions.append('evilbit.mark.not_seen')
+            elif conn0 is True and conn1 is False:
+                conditions.append('evilbit.connectivity.broken')
+            elif conn0 is False and conn1 is True:
+                conditions.append('evilbit.connectivity.transient')
+            elif conn0 is False and conn1 is False:
+                conditions.append('evilbit.connectivity.offline')
+
+        if self.args.connect == 'dnsudp':
+            if flows[0]['dns_response_valid'] and flows[1][
+                    'dns_response_valid']:
+                conditions.append('evilbit.connectivity.works')
+                if flows[1]['evil_bit_data_rev']:
+                    conditions.append('evilbit.mark.seen')
+                else:
+                    conditions.append('evilbit.mark.not_seen')
+            if flows[0]['dns_response_valid'] and not flows[1][
+                    'dns_response_valid']:
+                conditions.append('evilbit.connectivity.broken')
+            if not flows[0]['dns_response_valid'] and flows[1][
+                    'dns_response_valid']:
+                conditions.append('evilbit.connectivity.transient')
+            else:
+                conditions.append('evilbit.connectivity.offline')
+
+        return conditions
+
+    @classmethod
+    def register_args(cls, subparsers):
```

Comment:
In which case, that's a bug in forgespider. Can you file an issue for that and point out in that issue that the register_args function needs to be removed from evil bit when it's fixed.